src/clr-angular/forms/common/if-error/if-error.service.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable, OnDestroy} from "@angular/core";
+import {NgControl} from "@angular/forms";
+import {Observable, Subject, Subscription} from "rxjs";
+import {filter} from "rxjs/operators";
+
+import {NgControlService} from "../providers/ng-control.service";
+
+@Injectable()
+export class IfErrorService implements OnDestroy {
+    // Implement our own since it doesn't fire on events like blur, and return the control instead of a string
+    private _statusChanges: Subject<NgControl> = new Subject();
+    get statusChanges(): Observable<NgControl> {
+        return this._statusChanges.asObservable();
+    }
+
+    private subscriptions: Subscription[] = [];
+
+    constructor(private ngControlService: NgControlService) {
+        // Wait for the control to be available
+        this.subscriptions.push(this.ngControlService.controlChanges.subscribe(() => {
+            this.listenForChanges();
+        }));
+    }
+
+    // Subscribe to the status change events, only after touched and emit the control
+    private listenForChanges() {
+        this.subscriptions.push(
+            this.ngControlService.control.statusChanges.pipe(filter(() => this.ngControlService.control.touched))
+                .subscribe(() => {
+                    this._statusChanges.next(this.ngControlService.control);
+                }));
+    }
+
+    // Allows a control to push a status check upstream, such as on blur
+    checkStatus() {
```

Comment:
`checkStatus` makes me think that this method would return a `boolean` but instead, it emits the `NgControl`.  May be we should rename this method? May be `updateControlStatus`?